lib/src/lifecycle_module.dart

```diff
@@ -415,30 +405,19 @@ abstract class LifecycleModule extends SimpleModule
               : LifecycleState.suspended);
     }
 
-    if (!isLoaded) {
+    if (!(isLoaded || isLoading || isResuming || isSuspended)) {
       return _buildIllegalTransitionResponse(
           targetState: LifecycleState.suspended,
-          allowedStates: [LifecycleState.loaded]);
+          allowedStates: [LifecycleState.loaded, LifecycleState.resuming]);
     }
-
+    var previousTransition = _transition?.future;
+    _transition = new Completer<Null>();
     _state = LifecycleState.suspending;
-    final completer = new Completer<Null>();
-    _transitionFuture = completer.future;
-    _willSuspendController.add(this);
 
-    final suspendingChildren = _childModules.map((c) => c.suspend());
-    Future.wait(suspendingChildren).then((_) async {
-      await onSuspend();
-      _state = LifecycleState.suspended;
-      _didSuspendController.add(this);
-      _transitionFuture = null;
-      completer.complete();
-    }).catchError((Object error, StackTrace stackTrace) {
-      _didSuspendController.addError(error, stackTrace);
-      completer.completeError(error, stackTrace);
-    });
```

Comment:
This functionality has been [defined](https://github.com/Workiva/w_module/pull/81/files#diff-c586827f6717f60db651d466b5ab2b5fR637) as pure async private helper method.